lib/screens/new_game1/password_field.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:niira/models/view_models/new_game1.dart';
+import 'package:niira/utilities/validators.dart' as validators;
+
+class PasswordField extends StatefulWidget {
+  const PasswordField({Key key}) : super(key: key);
+
+  @override
+  _PasswordFieldState createState() => _PasswordFieldState();
+}
+
+class _PasswordFieldState extends State<PasswordField> {
+  NewGameViewModel1 _vm;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(30, 20, 30, 100),
+      child: TextFormField(
+        key: Key('new_game1_password_field'),
+        obscureText: true,
+        decoration: InputDecoration(
+            labelText: 'Password',
+            focusedBorder: OutlineInputBorder(
+              borderSide: BorderSide(color: Colors.black, width: 2.0),
+            ),
+            enabledBorder: OutlineInputBorder(
+              borderSide: BorderSide(color: Colors.black, width: 2.0),
+            ),
+            suffixIcon: Icon(Icons.visibility)),
+        validator: validators.oneToFifteenChars,
+        onChanged: (val) => setState(() => _vm?.password = val),
```

Comment:
what happens if vm is null?

Reply:
Good question! I guess it would break, although I'm not sure.  I believe it can never be null when the function runs because it's set in initState... I think maybe the null safety operator was to shut the analyzer up, I don't really remember.  Anyway it's a hangover from having the vm created in the parent widget, now it's in the service I can get rid of this nonsense.  Good pickup, thanks!

Reply:
alg!

Reply:
So I was wrong that it will break - if you put something like this into dartpad:

```Dart
class Classy {
  String stringy;  
}

void main() {
  Classy c;
  c?.stringy = 'hello';
  
  print(c);
}
```

I don't really get why it doesn't break though so even though the behaviour seems to be what we would want, I don't like it... who wrote this code anyway?  

When I try and make it nice though, it all breaks.  I would like to make the name and password widgets stateless and just get the VM out of the service as needed, but before I even get to stateless, if I just replace the `_vm?` with `context.read<GameService>().newGameViewModel1` the test fails, apparently because navigation didn't occur (presumably because the valid state was never obtained?). 

What would you like to do?  Take the code as is that works but is a bit strange or spend the time figuring out why changing it breaks?  I don't mind but I won't be able to spend any time on it tonight.

Reply:
Actually put the laptop down!  Rest up and get rid of that headache 🤒

Reply:
Aaaah I think I get it now!  

So the answer to your original question should have been:

If _vm is null in 

`_vm?.password = val`

then

`password = val`

just doesn't get called. 

It's the weird syntax of setters that got us ;-)  everything after the ? is just a function and if the object that the function is called on is null, the function doesn't get called.  If the line had have been: 

`_vm?.setPassword(val)`

I imagine neither of us would have been tricked.  

Well that was a great learning exercise anyway, thanks for the question!